File: 프로그래머스/2/49994. 방문 길이/방문 길이.py
def solution(dirs):
    s=set() # 내가 갔던 좌표 다 저장하면 될 것 같은데 현재좌표 다음좌표 & 다음좌표 내가 간 좌표 중복X
    d = {'U':(-1,0),
        'D':(1,0),
        'R':(0,1),
        'L':(0,-1)
        }
    x,y = 0,0
    
    for dir in dirs:
        nx,ny=  x+d[dir][0],y+d[dir][1]
        if -5<=nx<=5 and -5<=ny<=5:
            s.add((x,y,nx,ny))
            s.add((nx,ny,x,y))
            x,y=nx,ny
    
    return len(s)//2

File: 프로그래머스/2/49994. 방문 길이/방문 길이.js
function isVaildMove(nx,ny){
    //좌표평면 벗어났는지 쳌
    return nx >= -5 && nx<=5 && ny>=-5 &&ny<=5;
}

function updateLocation(x,y,dir){
    switch(dir){
        case "U":
            return [x,y+1];
        case "D":
            return [x,y-1];
        case "R":
            return [x+1,y];
        case "L":
            return [x-1,y];
            
    }
}

function solution(dirs) {
    let x =0;
    let y=0;
    
    const visited = new Set(); //겹치는 좌표는 1개로 처리하기 위함
    for(const dir of dirs){
        const [nx,ny] = updateLocation(x,y,dir)
        
        if (!isVaildMove(nx,ny)){
            continue;
        }
        
    visited.add(`${x}${y}${nx}${ny}`);
    visited.add(`${nx}${ny}${x}${y}`);
    
    [x,y] = [nx,ny];
    
    }
    return visited.size/2
}